core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts

```diff
@@ -197,14 +197,14 @@ export class ContributionAndReviewService {
       targetId: string, suggestionId: string, action: string,
       reviewMessage: string, commitMessage: string,
       onSuccess: (suggestionId: string) => void,
-      onFailure: (error) => void
+      onFailure: (error: Error) => void
   ): Promise<void> {
     const requestBody = {
       action: action,
       review_message: reviewMessage,
       commit_message: (
         action === AppConstants.ACTION_ACCEPT_SUGGESTION ?
-        commitMessage : null
+        commitMessage : ''
```

Comment:
Why is this changed to empty?

Reply:
Done

Reply:
Hm, that seems odd to me. Could you please add links and explanation to substantiate the claim you're making?

Reply:
see how we are dealing with commit message (means review message), we intialize them with empty string '' and if no review message, we submit suggestion with empty string

[link](https://github.com/oppia/oppia/blob/49a9c2087cb53c8b22b10748b93ab1769edfb8dd/core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.component.ts#L216)

Reply:
That's not a sufficient explanation. Take a look at (and perhaps provide a link to) the backend code that receives the request that is being sent. Does it expect null or a string in the rejection case?

Your statement that "when we accept or even reject any suggestion it always passes empty string, which further verify in checks" seems wrong in the "accept" case -- the conditional here should return the reviewMessage entered by the reviewer in that case. When you tested this manually, did you inspect the commit message in the request payload to verify that an empty string is sent?

Reply:
Ok, after investigation, i found commit message and review message are two different things, and we passes commit message as null in http request, when we directly accept solution without any suggestions. And that null value is further used to decide what kind of accept we do, like with suggestions or without suggestions. 

In rejection cases we always passes commit message with some value, means we do not need to wary about that.

Reply:
Could you please provide a link to the backend handler that accepts null for the commit message?

Reply:
[here](https://github.com/oppia/oppia/blob/2abb99616fb5f284d62605363ef36679992b2519/core/templates/pages/contributor-dashboard-page/services/contribution-and-review-backend-api.service.ts#L152) and requested body with null value is [here](https://github.com/oppia/oppia/blob/2abb99616fb5f284d62605363ef36679992b2519/core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts#L196)

Reply:
@JeeveshGarg, you are not analyzing the code properly.

This is a conditional statement of the form `A ? B : C`. Null is used when `action !== AppConstants.ACTION_ACCEPT_SUGGESTION`. In the rejection case, the suggestion is not applied, so there is no commit message (because there is no commit to make.

Reply:
Correct, i am talking about this same point, if that same conditions are applied in backend using `action`, so why we need to set commit_message as null, as it will not pass if action is under rejected.

Just let commit message be commit message.

OR

pass null directly in `rejectAndReviewNext` function for commit message in `translation-suggestion-review-modal.component`. Like passing commit string every time and then set to null in services every time.

Inbetween the state passing null is look confusing.

Reply:
Because semantically the commit_message _is_ None in the rejection case (i.e., there literally is no commit message -- we are not passing an empty string as the commit message), and it is good practice for code to represent the "real world" as much as possible.

For your suggestion, I took a look at the code and I do think it's slightly confusing. When this.generateCommitMessage() is called in rejectAndReviewNext(), it seems to generate something nonempty. That seems wrong, right -- and you're suggesting replacing that with null instead?

(Also, in general, I would suggest that you try to include links to code and code snippets in your replies. It is hard to understand sometimes what you are trying to explain, and the concrete code can help make it clearer.)

Reply:
Done now. Please see. Ya will take care from next time.